spec.md

```diff
@@ -286,6 +286,59 @@ circuit Foo :
   public module Foo enablelayer A :
 ```
 
+## Formal Tests
+
+Formal tests are names top-level constructs which allow for test-harnesses to be definied 
+as bounded model checking problems. Formal tests have 3 operands: 
+- A symbol that the test can be referred to with.  
+- The name of the test harness module.  
+- A bound for the bounded model checking problem, which refers to the number of cycles that 
+the module will be unrolled for in the formal test. This is a strictly positive integer.  
+
+```firrtl
+FIRRTL version 4.0.0
+circuit Foo :
+  public module Foo :
+    ; ...
+  formal testFoo of Foo, bound = 20
+```
+ 
+### Test Harness  
+
+A test harness is a public module that is typically only used for defining a test-bench.
+These are not instanciated and their inputs function as free (or symbolic) variables.
+Test harnesses typically instantiate a module, know as the Device Under Test (DUT), 
+connect a set of free variables to the DUT's inputs and then use those free variables to 
+check various properties of these DUT using assertions and assumptions.  
```

Comment:
Can you mention the keyword that is used to indicate the formal testharness? Is it `formal`?

Also can we standardize in this doc on Test harness vs testharness vs test-harness, same for testbench?

Reply:
test-harness + `formal` = formal test-bench. I'll standardize that!